pupil_src/shared_modules/gl_utils/utils.py

```diff
@@ -211,3 +222,89 @@ def __exit__(self, *exc):
         glPopMatrix()
         glMatrixMode(GL_MODELVIEW)
         glPopMatrix()
+
+
+class _Margins(T.NamedTuple):
+    left: int
+    top: int
+    right: int
+    bottom: int
+
+
+class _Rectangle(T.NamedTuple):
+    x: int
+    y: int
+    width: int
+    height: int
+
+    def intersection(self, other: "_Rectangle") -> T.Optional["_Rectangle"]:
+        in_min_x = max(self.x, other.x)
+        in_min_y = max(self.y, other.y)
+
+        in_max_x = min(self.x + self.width, other.x + other.width)
+        in_max_y = min(self.y + self.height, other.y + other.height)
+
+        if in_min_x < in_max_x and in_min_y < in_max_y:
+            return _Rectangle(
+                x=in_min_x,
+                y=in_min_y,
+                width=in_max_x - in_min_x,
+                height=in_max_y - in_min_y,
+            )
+        else:
+            return None
+
+
+def get_content_scale(window) -> float:
+    # TODO: Explicitly rename to `get_content_scale_x`
```

Comment:
@romanroibu here's a TODO left, did you plan to change something?

Reply:
Discussed this with @papr; We decided the name is good enough to hide the implementation details of how the scale value is calculated, and descriptive enough for the callers.